src/Umbraco.Web/Strategies/RelateOnTrashHandler.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System;
+using System.Linq;
+using Umbraco.Core;
+using Umbraco.Core.Auditing;
+using Umbraco.Core.Events;
+using Umbraco.Core.Models;
+using Umbraco.Core.Services;
+
+namespace Umbraco.Web.Strategies
+{
+    public sealed class RelateOnTrashHandler : ApplicationEventHandler
+    {
+        protected override void ApplicationStarting(UmbracoApplicationBase umbracoApplication, ApplicationContext applicationContext)
```

Comment:
This needs to be moved to ApplicationStarted, ApplicationStarting is a very specific event handler that is ONLY used to modify resolvers. See docs: http://our.umbraco.org/Documentation/Reference/Events-v6/Application-Startup#UseApplicationEventHandlertoregisterevents (though these should be updated to be more clear)